### DISCUSSION TITLE
Required key in KafkaTopic

### DISCUSSION THREAD

**Discussion by mathieux51** [strimzi/strimzi-kafka-operator#5216] [Q&A]
Is there anyway that `kind: KafkaTopic` could enforce the use of a Kafka key? Using [kaf CLI](https://github.com/birdayz/kaf) I can publish something like: 
```
echo '{"message": "foo"}' | kaf produce foo
echo '{"message": "foo"}' | kaf produce foo --key my-key
```
I would like that first one to tail somehow, possible?

**Comment by scholzj** [comment]
The `KafkaTopic` CR just creates the topic inside Kafka. But sending / receiving messages is fully in Kafka's responsibility. And I do not think Kafka supports anything like that.

**Reply by mathieux51** [reply]
Ok. Thank you 🙏
